build-system/tasks/check-links.js

```diff
@@ -51,68 +51,51 @@ function getMarkdownFiles() {
 async function checkLinks() {
   maybeUpdatePackages();
   const markdownFiles = getMarkdownFiles();
-  const linkCheckers = markdownFiles.map(function(markdownFile) {
-    return runLinkChecker(markdownFile);
-  });
-  return BBPromise.all(linkCheckers).then(function(allResults) {
-    let deadLinksFound = false;
-    const filesWithDeadLinks = [];
-    allResults.map(function(results, index) {
-      // Skip files that were deleted by the PR.
-      if (!fs.existsSync(markdownFiles[index])) {
-        return;
-      }
+  const allResults = await Promise.all(markdownFiles.map(runLinkChecker));
+
+  const filesWithDeadLinks = allResults
+    .map((results, index) => {
       let deadLinksFoundInFile = false;
-      results.forEach(function(result) {
+      for (const {link, status, statusCode} of results || []) {
         // Skip links to files that were introduced by the PR.
-        if (isLinkToFileIntroducedByPR(result.link)) {
-          return;
+        if (isLinkToFileIntroducedByPR(link)) {
+          continue;
         }
-        if (result.status === 'dead') {
-          deadLinksFound = true;
+        if (status === 'dead') {
           deadLinksFoundInFile = true;
-          log(`[${red('✖')}] ${result.link} (${red(result.statusCode)})`);
+          log(`[${red('✖')}]`, link, `(${red(statusCode)})`);
         } else if (!isTravisBuild()) {
-          log(`[${green('✔')}] ${result.link}`);
+          log(`[${green('✔')}]`, link);
         }
-      });
+      }
+      const filename = markdownFiles[index];
       if (deadLinksFoundInFile) {
-        filesWithDeadLinks.push(markdownFiles[index]);
-        log(
-          red('ERROR'),
-          'Possible dead link(s) found in',
-          magenta(markdownFiles[index])
-        );
-      } else {
-        log(
-          green('SUCCESS'),
-          'All links in',
-          magenta(markdownFiles[index]),
-          'are alive.'
-        );
+        log(red('ERROR'), 'Possible dead link(s) found in', magenta(filename));
+        return filename;
       }
-    });
-    if (deadLinksFound) {
-      log(
-        red('ERROR'),
-        'Please update dead link(s) in',
-        magenta(filesWithDeadLinks.join(',')),
-        'or whitelist them in build-system/tasks/check-links.js'
-      );
-      log(
-        yellow('NOTE'),
-        'If the link(s) above are not meant to resolve to a real webpage',
-        'surrounding them with backticks will exempt them from the link',
-        'checker.'
-      );
-      process.exitCode = 1;
-    } else {
-      log(
-        green('SUCCESS'),
-        'All links in all markdown files in this branch are alive.'
-      );
-    }
-  });
+      log(green('SUCCESS'), 'All links in', magenta(filename), 'are alive.');
+    })
+    .filter(filenameOrUndef => !!filenameOrUndef);
```

Comment:
What is this for?

Reply:
Judging by naming conventions, the variable is either a filename or undefined, and they are filtering out undefined. Though, if that is the case, `.filter(filenameOrUndefined => filenameOrUndefined)` would do fine

Reply:
Yup, just a simpler directional way to accomplish the same as keeping the local flag and pushing.